tools/validate_spec.py

```diff
@@ -103,19 +110,19 @@ def get_message_to_validate(message_path: Path, is_valid: bool) -> "OriginalMess
     try:
         identifier = ID(root_message_id)
     except AssertionError as e:
-        raise ValidationError(f'invalid identifier "{root_message_id}" : {str(e)}') from e
+        raise ValidationError(f'invalid identifier "{root_message_id}" : {e}') from e
```

Comment:
The `ID` class relies on assertions to check the identifier parts.

Reply:
I know.

> If that check is needed here, we should consider changing the implementation of ID in RecordFlux.

1. Please analyze if checking the validity of the identifier is really needed here.
2. If this is the case, open an issue for RecordFlux to change the implementation of `ID`, so that an application-specific exception is raised when the identifier parts are malformed.

Reply:
Are the CI tests run with assertions enabled ot disabled? 
I added an alternative for checking the validity of an identifier without relying on assertions in the ID class. However, i left the try/except block for catching the AssertionError in, in case the tool is run with assertions enabled, so that a user is not confronted with an assertion error directly. 5a9f625

Reply:
I don't think that's a good solution. The point @treiher was trying to make is that `AssertionError` should never be used to signal an expected user error (contrary to an unexpected programming error). While you worked around the issue, the problem persists in RecordFlux. The right way to fix this issue is to open a ticket in RecordFlux, resolve the problem in RecordFlux and after that change your validation tool to rely on the proper exception (probably RecordFluxError).

Reply:
Waiting for https://github.com/Componolit/RecordFlux/pull/541 to be merged

Reply:
Changed in d75b93a